// vue-nav-ui/dist/vue-nav-ui.ssr.js
'use strict';var vue=require('vue');function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArrayLimit(arr, i) {
  if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return;
  var _arr = [];
  var _n = true;
  var _d = false;
  var _e = undefined;

  try {
    for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];

  return arr2;
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}var __default__ = vue.defineComponent({
  name: "Nav",
  props: {
    navLinks: {
      type: Array,
      required: true
    },
    navConfig: {
      type: Object,
      required: true
    },
    btnConfig: {
      type: Object,
      required: true
    }
  },
  data: function data() {
    return {
      open: false
    };
  }
});

var __injectCSSVars__ = function __injectCSSVars__() {
  vue.useCssVars(function (_ctx) {
    var _ref;

    return _ref = {
      "6e6856b7": _ctx.navConfig.navBorderRadius,
      "632f725e": _ctx.navConfig.navBg,
      "1a6b3613": _ctx.navConfig.linkFont,
      "32d13e1f": _ctx.navConfig.linkColor,
      "906b5a2e": _ctx.btnConfig.btnBorderRadius,
      "7ba3a510": _ctx.btnConfig.btnBg,
      "ec69a9ea": _ctx.btnConfig.btnTextColor
    }, _defineProperty(_ref, "1a6b3613", _ctx.navConfig.linkFont), _defineProperty(_ref, "6950a08f", _ctx.btnConfig.btnBorderWidth), _defineProperty(_ref, "6839a22c", _ctx.btnConfig.btnBorderColor), _defineProperty(_ref, "632f725e", _ctx.navConfig.navBg), _defineProperty(_ref, "632f725e", _ctx.navConfig.navBg), _ref;
  });
};

var __setup__ = __default__.setup;
__default__.setup = __setup__ ? function (props, ctx) {
  __injectCSSVars__();

  return __setup__(props, ctx);
} : __injectCSSVars__;var _withId = /*#__PURE__*/vue.withScopeId("data-v-6d7dccb2");

vue.pushScopeId("data-v-6d7dccb2");

var _hoisted_1 = {
  class: "nav"
};
var _hoisted_2 = {
  width: "28",
  height: "28",
  viewBox: "0 0 28 28",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};

var _hoisted_3 = /*#__PURE__*/vue.createVNode("div", {
  class: "excess"
}, null, -1);

vue.popScopeId();

var render = /*#__PURE__*/_withId(function (_ctx, _cache, $props, $setup, $data, $options) {
  var _component_router_link = vue.resolveComponent("router-link");

  return vue.openBlock(), vue.createBlock(vue.Fragment, null, [vue.createVNode("div", {
    class: {
      container: _ctx.navConfig.whitespace,
      open: _ctx.open
    }
  }, [vue.createVNode("div", _hoisted_1, [vue.renderSlot(_ctx.$slots, "default"), vue.createVNode("div", {
    class: ["links", {
      open: _ctx.open,
      bottom: _ctx.navConfig.responsivePosition == 'bottom',
      top: _ctx.navConfig.responsivePosition == 'top'
    }]
  }, [(vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.navLinks, function (navName, idx) {
    return vue.openBlock(), vue.createBlock(_component_router_link, {
      class: "link",
      key: idx,
      to: navName.path
    }, {
      default: _withId(function () {
        return [vue.createTextVNode(vue.toDisplayString(navName.name), 1)];
      }),
      _: 2
    }, 1032, ["to"]);
  }), 128)), _ctx.btnConfig.btnLink ? (vue.openBlock(), vue.createBlock("a", {
    key: 0,
    class: "linkBtn",
    href: _ctx.btnConfig.btnUrl
  }, [vue.createVNode("button", null, [vue.renderSlot(_ctx.$slots, "btnAppend"), vue.createVNode("p", null, vue.toDisplayString(_ctx.btnConfig.btnText), 1), vue.renderSlot(_ctx.$slots, "btnPrepend")])], 8, ["href"])) : vue.createCommentVNode("", true)], 2), vue.createVNode("div", {
    class: [{
      open: _ctx.open
    }, "toggle"],
    onClick: _cache[1] || (_cache[1] = function ($event) {
      return _ctx.open = !_ctx.open;
    })
  }, [(vue.openBlock(), vue.createBlock("svg", _hoisted_2, [vue.createVNode("rect", {
    x: "28",
    y: "12",
    width: "4",
    height: "28",
    rx: "2",
    transform: "rotate(90 28 12)",
    fill: _ctx.navConfig.linkColor
  }, null, 8, ["fill"]), vue.createVNode("rect", {
    x: "16",
    y: "28",
    width: "4",
    height: "28",
    rx: "2",
    transform: "rotate(-180 16 28)",
    fill: _ctx.navConfig.linkColor
  }, null, 8, ["fill"])]))], 2), _ctx.btnConfig.btnLink ? (vue.openBlock(), vue.createBlock("a", {
    key: 0,
    class: "navBtn",
    href: _ctx.btnConfig.btnUrl
  }, [vue.createVNode("button", null, [vue.renderSlot(_ctx.$slots, "btnAppend"), vue.createVNode("p", null, vue.toDisplayString(_ctx.btnConfig.btnText), 1), vue.renderSlot(_ctx.$slots, "btnPrepend")])], 8, ["href"])) : vue.createCommentVNode("", true)])], 2), _hoisted_3], 64);
});function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}var css_248z = "\n.excess[data-v-6d7dccb2] {\n  height: 122px;\n  width: 100%;\n  display: none;\n}\n.container[data-v-6d7dccb2] {\n  padding: 20px;\n}\n.nav[data-v-6d7dccb2] {\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  width: 1400px;\n  margin: auto;\n  border-radius: var(--6e6856b7);\n  padding: 16px;\n  background: var(--632f725e);\n}\n@media (max-width: 1400px) {\n.nav[data-v-6d7dccb2] {\n    width: 100%;\n}\n}\n.link[data-v-6d7dccb2] {\n  text-decoration: none;\n  margin: 0 20px;\n  font-family: var(--1a6b3613);\n  color: var(--32d13e1f);\n}\nbutton[data-v-6d7dccb2] {\n  border-radius: var(--906b5a2e);\n  background: var(--7ba3a510);\n  color: var(--ec69a9ea);\n  font-family: var(--1a6b3613);\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  border: none;\n  padding: 10px 20px;\n  border: var(--6950a08f) solid var(--6839a22c);\n  cursor: pointer;\n  outline: none;\n  font-size: 18px;\n}\nbutton p[data-v-6d7dccb2] {\n  margin: 0 10px;\n}\n.right[data-v-6d7dccb2] {\n  flex-direction: row;\n}\n.linkBtn[data-v-6d7dccb2] {\n  display: none;\n}\n.left[data-v-6d7dccb2] {\n  flex-direction: row-reverse;\n}\n.toggle[data-v-6d7dccb2] {\n  display: none;\n}\n@media (max-width: 790px) {\n.excess[data-v-6d7dccb2] {\n    display: block;\n}\n.container[data-v-6d7dccb2] {\n    height: 100vh;\n    position: fixed;\n    left: 0;\n    pointer-events: none;\n    top: 0;\n    width: 100%;\n    transition: all 0.3s;\n}\n.container.open[data-v-6d7dccb2] {\n    pointer-events: all;\n    background: rgba(0, 0, 0, 0.337);\n    backdrop-filter: blur(4px);\n}\n.toggle[data-v-6d7dccb2] {\n    display: block;\n    padding: 10px;\n    background: var(--632f725e);\n    border-radius: 100%;\n    width: 50px;\n    height: 50px;\n    cursor: pointer;\n    transform: rotate(0deg);\n    pointer-events: all;\n    transition: all 0.3s;\n}\n.toggle svg[data-v-6d7dccb2] {\n    width: 100%;\n    height: 100%;\n}\n.toggle.open[data-v-6d7dccb2] {\n    transform: rotate(45deg);\n}\n.nav[data-v-6d7dccb2] {\n    position: relative;\n    background: none;\n    justify-content: space-between;\n}\n.links[data-v-6d7dccb2] {\n    position: fixed;\n    left: 50%;\n    background: var(--632f725e);\n    display: flex;\n    flex-direction: column;\n    padding: 40px 30px;\n    width: 90%;\n    margin-top: 80px;\n    border-radius: 20px;\n    transform: translateY(50px) translateX(-50%);\n    pointer-events: none;\n    transition: all 0.3s;\n    opacity: 0;\n}\n.links.open[data-v-6d7dccb2] {\n    opacity: 1;\n    transform: translateY(0px) translateX(-50%);\n    pointer-events: all;\n}\n.links.bottom[data-v-6d7dccb2] {\n    bottom: 0;\n    border-radius: 20px 20px 0px 0px;\n}\n.links.top[data-v-6d7dccb2] {\n    top: 40px;\n}\n.links .link[data-v-6d7dccb2] {\n    margin: 15px 0;\n}\n.navBtn[data-v-6d7dccb2] {\n    display: none;\n}\n.linkBtn[data-v-6d7dccb2] {\n    display: block;\n    margin-top: 20px;\n}\n}\n";
styleInject(css_248z);__default__.render = render;
__default__.__scopeId = "data-v-6d7dccb2";// Import vue component

// Default export is installable instance of component.
// IIFE injects install function into component, allowing component
// to be registered via Vue.use() as well as Vue.component(),
var component = /*#__PURE__*/(function () {
  // Assign InstallableComponent type
  var installable = __default__; // Attach install function executed by Vue.use()

  installable.install = function (app) {
    app.component('VueNavUi', installable);
  };

  return installable;
})(); // It's possible to expose named exports when writing components that can
// also be used as directives, etc. - eg. import { RollupDemoDirective } from 'rollup-demo';
// export const RollupDemoDirective = directive;
var namedExports=/*#__PURE__*/Object.freeze({__proto__:null,'default': component});// only expose one global var, with named exports exposed as properties of
// that global var (eg. plugin.namedExport)

Object.entries(namedExports).forEach(function (_ref) {
  var _ref2 = _slicedToArray(_ref, 2),
      exportName = _ref2[0],
      exported = _ref2[1];

  if (exportName !== 'default') component[exportName] = exported;
});module.exports=component;
