// vue-nav-ui/dist/vue-nav-ui.esm.js
import { defineComponent, useCssVars, pushScopeId, popScopeId, resolveComponent, openBlock, createBlock, Fragment, createVNode, renderSlot, renderList, createTextVNode, toDisplayString, createCommentVNode, withScopeId } from 'vue';

const __default__ = defineComponent({
  name: "Nav",
  props: {
    navLinks: {
      type: Array,
      required: true
    },
    navConfig: {
      type: Object,
      required: true
    },
    btnConfig: {
      type: Object,
      required: true
    }
  },

  data() {
    return {
      open: false
    };
  }

});

const __injectCSSVars__ = () => {
  useCssVars(_ctx => ({
    "6e6856b7": _ctx.navConfig.navBorderRadius,
    "632f725e": _ctx.navConfig.navBg,
    "1a6b3613": _ctx.navConfig.linkFont,
    "32d13e1f": _ctx.navConfig.linkColor,
    "906b5a2e": _ctx.btnConfig.btnBorderRadius,
    "7ba3a510": _ctx.btnConfig.btnBg,
    "ec69a9ea": _ctx.btnConfig.btnTextColor,
    "1a6b3613": _ctx.navConfig.linkFont,
    "6950a08f": _ctx.btnConfig.btnBorderWidth,
    "6839a22c": _ctx.btnConfig.btnBorderColor,
    "632f725e": _ctx.navConfig.navBg,
    "632f725e": _ctx.navConfig.navBg
  }));
};

const __setup__ = __default__.setup;
__default__.setup = __setup__ ? (props, ctx) => {
  __injectCSSVars__();

  return __setup__(props, ctx);
} : __injectCSSVars__;

const _withId = /*#__PURE__*/withScopeId("data-v-6d7dccb2");

pushScopeId("data-v-6d7dccb2");

const _hoisted_1 = {
  class: "nav"
};
const _hoisted_2 = {
  width: "28",
  height: "28",
  viewBox: "0 0 28 28",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};

const _hoisted_3 = /*#__PURE__*/createVNode("div", {
  class: "excess"
}, null, -1);

popScopeId();

const render = /*#__PURE__*/_withId((_ctx, _cache, $props, $setup, $data, $options) => {
  const _component_router_link = resolveComponent("router-link");

  return openBlock(), createBlock(Fragment, null, [createVNode("div", {
    class: {
      container: _ctx.navConfig.whitespace,
      open: _ctx.open
    }
  }, [createVNode("div", _hoisted_1, [renderSlot(_ctx.$slots, "default"), createVNode("div", {
    class: ["links", {
      open: _ctx.open,
      bottom: _ctx.navConfig.responsivePosition == 'bottom',
      top: _ctx.navConfig.responsivePosition == 'top'
    }]
  }, [(openBlock(true), createBlock(Fragment, null, renderList(_ctx.navLinks, (navName, idx) => {
    return openBlock(), createBlock(_component_router_link, {
      class: "link",
      key: idx,
      to: navName.path
    }, {
      default: _withId(() => [createTextVNode(toDisplayString(navName.name), 1)]),
      _: 2
    }, 1032, ["to"]);
  }), 128)), _ctx.btnConfig.btnLink ? (openBlock(), createBlock("a", {
    key: 0,
    class: "linkBtn",
    href: _ctx.btnConfig.btnUrl
  }, [createVNode("button", null, [renderSlot(_ctx.$slots, "btnAppend"), createVNode("p", null, toDisplayString(_ctx.btnConfig.btnText), 1), renderSlot(_ctx.$slots, "btnPrepend")])], 8, ["href"])) : createCommentVNode("", true)], 2), createVNode("div", {
    class: [{
      open: _ctx.open
    }, "toggle"],
    onClick: _cache[1] || (_cache[1] = $event => _ctx.open = !_ctx.open)
  }, [(openBlock(), createBlock("svg", _hoisted_2, [createVNode("rect", {
    x: "28",
    y: "12",
    width: "4",
    height: "28",
    rx: "2",
    transform: "rotate(90 28 12)",
    fill: _ctx.navConfig.linkColor
  }, null, 8, ["fill"]), createVNode("rect", {
    x: "16",
    y: "28",
    width: "4",
    height: "28",
    rx: "2",
    transform: "rotate(-180 16 28)",
    fill: _ctx.navConfig.linkColor
  }, null, 8, ["fill"])]))], 2), _ctx.btnConfig.btnLink ? (openBlock(), createBlock("a", {
    key: 0,
    class: "navBtn",
    href: _ctx.btnConfig.btnUrl
  }, [createVNode("button", null, [renderSlot(_ctx.$slots, "btnAppend"), createVNode("p", null, toDisplayString(_ctx.btnConfig.btnText), 1), renderSlot(_ctx.$slots, "btnPrepend")])], 8, ["href"])) : createCommentVNode("", true)])], 2), _hoisted_3], 64);
});

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = "\n.excess[data-v-6d7dccb2] {\n  height: 122px;\n  width: 100%;\n  display: none;\n}\n.container[data-v-6d7dccb2] {\n  padding: 20px;\n}\n.nav[data-v-6d7dccb2] {\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  width: 1400px;\n  margin: auto;\n  border-radius: var(--6e6856b7);\n  padding: 16px;\n  background: var(--632f725e);\n}\n@media (max-width: 1400px) {\n.nav[data-v-6d7dccb2] {\n    width: 100%;\n}\n}\n.link[data-v-6d7dccb2] {\n  text-decoration: none;\n  margin: 0 20px;\n  font-family: var(--1a6b3613);\n  color: var(--32d13e1f);\n}\nbutton[data-v-6d7dccb2] {\n  border-radius: var(--906b5a2e);\n  background: var(--7ba3a510);\n  color: var(--ec69a9ea);\n  font-family: var(--1a6b3613);\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  border: none;\n  padding: 10px 20px;\n  border: var(--6950a08f) solid var(--6839a22c);\n  cursor: pointer;\n  outline: none;\n  font-size: 18px;\n}\nbutton p[data-v-6d7dccb2] {\n  margin: 0 10px;\n}\n.right[data-v-6d7dccb2] {\n  flex-direction: row;\n}\n.linkBtn[data-v-6d7dccb2] {\n  display: none;\n}\n.left[data-v-6d7dccb2] {\n  flex-direction: row-reverse;\n}\n.toggle[data-v-6d7dccb2] {\n  display: none;\n}\n@media (max-width: 790px) {\n.excess[data-v-6d7dccb2] {\n    display: block;\n}\n.container[data-v-6d7dccb2] {\n    height: 100vh;\n    position: fixed;\n    left: 0;\n    pointer-events: none;\n    top: 0;\n    width: 100%;\n    transition: all 0.3s;\n}\n.container.open[data-v-6d7dccb2] {\n    pointer-events: all;\n    background: rgba(0, 0, 0, 0.337);\n    backdrop-filter: blur(4px);\n}\n.toggle[data-v-6d7dccb2] {\n    display: block;\n    padding: 10px;\n    background: var(--632f725e);\n    border-radius: 100%;\n    width: 50px;\n    height: 50px;\n    cursor: pointer;\n    transform: rotate(0deg);\n    pointer-events: all;\n    transition: all 0.3s;\n}\n.toggle svg[data-v-6d7dccb2] {\n    width: 100%;\n    height: 100%;\n}\n.toggle.open[data-v-6d7dccb2] {\n    transform: rotate(45deg);\n}\n.nav[data-v-6d7dccb2] {\n    position: relative;\n    background: none;\n    justify-content: space-between;\n}\n.links[data-v-6d7dccb2] {\n    position: fixed;\n    left: 50%;\n    background: var(--632f725e);\n    display: flex;\n    flex-direction: column;\n    padding: 40px 30px;\n    width: 90%;\n    margin-top: 80px;\n    border-radius: 20px;\n    transform: translateY(50px) translateX(-50%);\n    pointer-events: none;\n    transition: all 0.3s;\n    opacity: 0;\n}\n.links.open[data-v-6d7dccb2] {\n    opacity: 1;\n    transform: translateY(0px) translateX(-50%);\n    pointer-events: all;\n}\n.links.bottom[data-v-6d7dccb2] {\n    bottom: 0;\n    border-radius: 20px 20px 0px 0px;\n}\n.links.top[data-v-6d7dccb2] {\n    top: 40px;\n}\n.links .link[data-v-6d7dccb2] {\n    margin: 15px 0;\n}\n.navBtn[data-v-6d7dccb2] {\n    display: none;\n}\n.linkBtn[data-v-6d7dccb2] {\n    display: block;\n    margin-top: 20px;\n}\n}\n";
styleInject(css_248z);

__default__.render = render;
__default__.__scopeId = "data-v-6d7dccb2";

// Import vue component

// Default export is installable instance of component.
// IIFE injects install function into component, allowing component
// to be registered via Vue.use() as well as Vue.component(),
var entry_esm = /*#__PURE__*/(() => {
  // Assign InstallableComponent type
  const installable = __default__; // Attach install function executed by Vue.use()

  installable.install = app => {
    app.component('VueNavUi', installable);
  };

  return installable;
})(); // It's possible to expose named exports when writing components that can
// also be used as directives, etc. - eg. import { RollupDemoDirective } from 'rollup-demo';
// export const RollupDemoDirective = directive;

export default entry_esm;
